planner/optimize.go

```diff
@@ -75,6 +75,10 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 	if bindRecord == nil {
 		return bestPlan, names, nil
 	}
+	if sctx.GetSessionVars().SelectLimit != math.MaxUint64 {
+		logutil.BgLogger().Info("sql_select_limit is set, so we will not consider plan bindings")
```

Comment:
Could this log be shown as an INFO or a WARNING when calling `explain` or `explain analyze`? So that the server log can be avoided spamming.